template.yaml

```diff
@@ -124,6 +125,141 @@ Resources:
       PointInTimeRecoverySpecification:
         PointInTimeRecoveryEnabled: true
 
+  #===============================DLOQs===================================================================
+
+  DynamodbEventFanoutStreamRecordsDLQ:
+    Type: "AWS::SQS::Queue"
+  UpdateDoiStatusDLQ:
+    Type: "AWS::SQS::Queue"
+  DeletePublicationEventProducerDlq:
+    Type: "AWS::SQS::Queue"
+  PublicationFanoutHandlerDLQ:
+    Type: "AWS::SQS::Queue"
+  DeleteDraftPublicationHandlerDLQ:
+    Type: "AWS::SQS::Queue"
+  DoiRequestEventProducerDLQ:
+    Type: "AWS::SQS::Queue"
+
+
+  #==============================ROLES=======================================================================
+
+  DeleteDraftPublicationLambdaRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service: lambda.amazonaws.com
+            Action: sts:AssumeRole
+      Path: /
+      Policies:
+        - PolicyName: Events
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Effect: Allow
+                Action:
+                  - events:PutEvents
+                Resource: !Ref EventBusArn
+              - Effect: Allow
+                Action:
+                  - sqs:SendMessage
+                  - sqs:ReceiveMessage
+                  - sqs:DeleteMessage
+                Resource: !GetAtt DeleteDraftPublicationHandlerDLQ.Arn
+
+  UpdateDoiStatusLambdaRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service: lambda.amazonaws.com
+            Action: sts:AssumeRole
+      Path: /
+      Policies:
+        - PolicyName: Events
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Effect: Allow
+                Action:
+                  - sqs:SendMessage
+                  - sqs:ReceiveMessage
+                  - sqs:DeleteMessage
+                Resource: !GetAtt UpdateDoiStatusDLQ.Arn
```

Comment:
Create another policy for this or split the "EventsLambdaPermissions" policy to SQS and eventbridge